pyclang/cli_ext.py

```diff
@@ -21,6 +21,11 @@
     type=os.path.realpath,
     help='where the log files will be write to. will use stdout if not specified',
 )
+common_args.add_argument(
+    '--fail-on-error',
+    action='store_true',
+    help='Raise an exception if run-clang-tidy.py script fails. By default end successfully with a summary of found errors.',
```

Comment:
"Raise an exception" is a little too implementation specific. Could we have something like: "Exit with code based on the results of the code analysis. By default, exit code reflects the success of running the tool only." 

I would also recommend to rename the argument. What do you think about `--analysis-exit-code`?

Reply:
I will change the help, thanks for the suggestion.
About the option name:  `--analysis-exit-code` seems a bit unclear what will happened to the exit code, I would prefer something like `--preserve-analysis-exit-code` but that is too long. What about `--preserve-exit-code`?

Reply:
`--preserve-exit-code` or `--exit-code` also sound good to me.

Note that `git diff` for example have just `--exit-code`.

Reply:
okay, so `--exit-code` it is, thanks.